### PR TITLE
Add Dockerfile to generate a base image for plugins

### PR DESCRIPTION
**Pull request by cdeepakroy**
Added a Dockerfile to generate a base image that can be used by all slicer_cli_web plugins.

This Dockerfile does the following:
* apt-get install system pre-requisites
* Download and Install miniconda
* Download and Install CMake
* git clone and build ITK 4.10
* git clone and build SlicerExecutionModel
* Install ctk_cli=1.3.1 from my conda channel
* Copy in files of slicer_cli_web

### CODE REVIEW

**Comment by cdeepakroy** [comment]
@jcfr Can you please take a look?

**Comment by cdeepakroy** [comment]
This would simplify the Dockerfile of plugins a lot. See my corresponding [PR](https://github.com/cdeepakroy/slicer_cli_web_plugin/pull/5) on the slicer_cli_web_plugin.

**Comment by jcfr** [comment]
Looks good :+1: 

I suggest you squash the commits and add a commit message like this one:

```
Add Dockerfile for generating base image for plugin

This commit backports most of the work originally done in `slicer_cli_web_plugin`
project so that the created Docker image can serve as a generic base for 
other Slicer CLI web plugins (either python or C++).

In a nutshell, this `Dockerfile` does the following:

* `apt-get install` system pre-requisites
* Download and Install miniconda
* Download and Install CMake
* git clone and build ITK 4.10
* git clone and build SlicerExecutionModel
* Install ctk_cli=1.3.1 from my conda channel
* Copy in files of slicer_cli_web


It also refactors the code to remove the dependency on Girder. The only change
required was to remove the use of ``girder.logger.Exception``.

Co-authored-by: Jean-Christophe Fillion-Robin <jchris.fillionr@kitware.com>
```

with you as author. 

Thanks
Jc

File: server/cli_list_entrypoint.py [hunk 1]
```diff
@@ -5,8 +5,6 @@
 import subprocess
 import textwrap as _textwrap
 
-from girder import logger
-
```
Comment (manthey) [inline server/cli_list_entrypoint.py]: We can actually do better than just removing this:
```
try:
   from girder import logger
except ImportError:
   import logging as logger
```
and then leave the log statements in place.

**Pull request merged**